File: src/main/java/com/distribution/modules/dis/service/DisProfiParamService.java
package com.distribution.modules.dis.service;

import com.distribution.modules.dis.entity.DisMemberInfoEntity;
import com.distribution.modules.dis.entity.DisProfiParam;

import java.util.List;
import java.util.Map;


/**
 * 分润参数设置
 *
 * @author huchunliang
 * @email davichi2009@gmail.com
 * @date 2018-05-22
 */
public interface DisProfiParamService {
    /**
     * 根据ID查询
     *
     * @param id
     * @return
     */
    DisProfiParam queryObject(String id);

    /**
     * 查询列表
     *
     * @param map
     * @return
     */
    List<DisProfiParam> queryList(Map<String, Object> map);

    /**
     * 保存
     *
     * @param disProfiParam
     * @throws Exception
     */
    void save(DisProfiParam disProfiParam) throws Exception;

    /**
     * 更新
     *
     * @param disProfiParam
     * @throws Exception
     */
    void update(DisProfiParam disProfiParam) throws Exception;

    /**
     * 删除
     *
     * @param id
     * @throws Exception
     */
    void delete(String id) throws Exception;

    /**
     * 批量删除
     *
     * @param ids
     * @throws Exception
     */
    void deleteBatch(String[] ids) throws Exception;

    /**
     * 分润逻辑
     *
     * @param member     会员信息
     * @param money      金额
     * @param isReward 是否是会员推荐奖励
     */
    void doFeeSplitting(DisMemberInfoEntity member, Double money,boolean isReward) throws Exception;
}

File: src/main/java/com/distribution/modules/dis/service/DisProfiParamService.kt
package com.distribution.modules.dis.service

import com.distribution.modules.dis.entity.DisMemberInfoEntity
import com.distribution.modules.dis.entity.DisProfiParam


/**
 * 分润参数设置
 *
 * @author huchunliang
 * @email davichi2009@gmail.com
 * @date 2018-05-22
 */
interface DisProfiParamService {
    /**
     * 根据ID查询
     *
     * @param id
     * @return
     */
    fun queryObject(id: String): DisProfiParam

    /**
     * 查询列表
     *
     * @param map
     * @return
     */
    fun queryList(map: Map<String, Any>): List<DisProfiParam>

    /**
     * 保存
     *
     * @param disProfiParam
     * @throws Exception
     */
    @Throws(Exception::class)
    fun save(disProfiParam: DisProfiParam)

    /**
     * 更新
     *
     * @param disProfiParam
     * @throws Exception
     */
    @Throws(Exception::class)
    fun update(disProfiParam: DisProfiParam)

    /**
     * 删除
     *
     * @param id
     * @throws Exception
     */
    @Throws(Exception::class)
    fun delete(id: String)

    /**
     * 批量删除
     *
     * @param ids
     * @throws Exception
     */
    @Throws(Exception::class)
    fun deleteBatch(ids: Array<String>)

    /**
     * 分润逻辑
     *
     * @param member   会员信息
     * @param money    金额
     * @param isReward 是否是会员推荐奖励
     */
    @Throws(Exception::class)
    fun doFeeSplitting(member: DisMemberInfoEntity, money: Double, isReward: Boolean)
}
